lib/ansible/galaxy/role.py

```diff
@@ -45,6 +45,20 @@
 display = Display()
 
 
+def is_rel_path_outside_archive(archive_parent_dir : str, path : str, path_must_exist : bool = False) -> bool:
+    """
+    Check if the path (relative to the archive) resolves to a path in the archive, and optionally validate the path exists.
+    """
+    try:
+        # Get the absolute paths
+        archive_dir = os.path.abspath(archive_parent_dir)
+        path = os.path.abspath(os.path.join(archive_parent_dir, path))
+
+        return not (path.startswith(archive_dir) and (not path_must_exist or os.path.exists(path)))
+    except OSError:
+        return True
```

Comment:
lets not rewrite things, you can use `from lib/ansible/utils/path.py import is_subpath`

Reply:
didn't know it existed, thanks for pointing it out